firewood/src/merkle/stream.rs

```diff
@@ -396,6 +396,140 @@ impl<'a, S: ShaleStore<Node> + Send + Sync, T> Stream for MerkleKeyValueStream<'
     }
 }
 
+enum PathIteratorState<'a> {
+    Iterating {
+        /// The key, as nibbles, of the node at [address], without the
+        /// node's partial path (if any) at the end.
+        /// Invariant: If this node has a parent, the parent's key is a
+        /// prefix of the key we're seeking.
+        /// Note the node at [address] may not have a key which is a
+        /// prefix of the key we're seeking.
```

Comment:
I would use the term "iterating over" instead of "seeking". 

The note is also kind of confusing. I think it would be better to describe the situation where the key is not a prefix of the key we're iterating over. When do we run into this case? I would expect the path-iterator to strictly return nodes that *are* on the path. 

Is it because the address is the address of the next node to check?

Reply:
updated the comment, lmk what you think

Reply:
To be honest, I find the `Note` part of the comment a little confusing, but it doesn't matter, I understand the code!